query/sortUser.go

```diff
@@ -0,0 +1,103 @@
+package query
+
+import (
+	"github.com/ONSdigital/dp-identity-api/models"
+	"sort"
+	"strings"
+)
+
+type lessFunc func(p1, p2 *models.UserParams) bool
+
+type multiSorter struct {
+	changes []models.UserParams
+	less    []lessFunc
+}
+
+func (ms *multiSorter) Sort(changes []models.UserParams) {
+	ms.changes = changes
+	sort.Sort(ms)
+}
+
+func OrderedBy(less ...lessFunc) *multiSorter {
+	return &multiSorter{
+		less: less,
+	}
+}
+func (ms *multiSorter) Len() int {
+	return len(ms.changes)
+}
+func (ms *multiSorter) Swap(i, j int) {
+	ms.changes[i], ms.changes[j] = ms.changes[j], ms.changes[i]
+}
+func (ms *multiSorter) Less(i, j int) bool {
+	p, q := &ms.changes[i], &ms.changes[j]
+	var k int
+	for k = 0; k < len(ms.less)-1; k++ {
+		less := ms.less[k]
+		switch {
+		case less(p, q):
+			return true
+		case less(q, p):
+			return false
+		}
+	}
+	return ms.less[k](p, q)
+}
+
+func SortBy(paramsSlice string, arr []models.UserParams) {
```

Comment:
I'm not sure about how descriptive the 'paramsSlice' is. 

Godoc would be useful here too (and on the above docs)